tests/evolution/test_evolution_config_validation.py

```
@@ -0,0 +1,151 @@
+        assert hasattr(config, 'semantic_batch_size')
+        assert hasattr(config, 'semantic_cache_ttl')
+
+    def test_validation_error_messages(self):
+        """Test that validation provides clear error messages."""
+        config = EvolutionConfig(population_size=1)
+        
+        # For now, validate() returns bool, but we'll enhance it later
+        # to return validation errors
+        assert config.validate() is False
+
+    def test_combined_valid_config(self):
```

Ruff (0.12.2)

Missing return type annotation for public function `test_combined_valid_config`

Add return type annotation: `None`

(ANN201)